tsconfig.json

```diff
@@ -4,7 +4,7 @@
         "module": "esnext",
         "moduleResolution": "node",
 
-        "lib": ["es2015"],
+        "lib": ["es2015", "dom"],
```

Comment:
I had to add this due to https://github.com/DefinitelyTyped/DefinitelyTyped/issues/12044 😕

Reply:
idk if that's actually a bug. For example nodejs doesn't have `XMLHttpRequest`, only browsers do, and "dom" basically means browsers.

I think I'd like to avoid this though, since this library should work outside the dom. I think you can what this guy suggests: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/12044#issuecomment-256067473

Just add `interface XMLHttpRequest {}` to the spec file somewhere with a comment explaining it.

Or the other option is adding a separate `tsconfig.json` just for tests. You can create `tsconfig.spec.json` and "extend" the main `tsconfig.json` but override the `lib` property, see https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#configuration-inheritance-with-extends

Reply:
Actually maybe just do the same thing that we do for `window`, `document` and `Blob` in the tests. At the top of `imuter.spec.ts`:

```
declare var window: any;
declare var document: any;
declare var Blob: any;
```

Does adding `XMLHttpRequest` to that list work?